chasm/lib/worker/handler_test.go

```diff
@@ -0,0 +1,105 @@
+package worker
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	workerpb "go.temporal.io/api/worker/v1"
+	"go.temporal.io/server/chasm"
+	workerstatepb "go.temporal.io/server/chasm/lib/worker/gen/workerpb/v1"
+	"go.temporal.io/server/common/dynamicconfig"
+	"go.temporal.io/server/common/log"
+	"go.temporal.io/server/common/testing/testvars"
+	"go.uber.org/mock/gomock"
+	"google.golang.org/protobuf/types/known/durationpb"
+)
+
+const (
+	testWorkerInstanceKey = "test-worker-1"
+)
+
+func TestRecordHeartbeatHandler(t *testing.T) {
```

Comment:
Make this a functional test, all of these mocks don't prove that this API actually works.